lib/spack/spack/solver/concretize.lp

```diff
@@ -48,11 +32,21 @@ version_declared(Package, Version) :- version_declared(Package, Version, _).
 
 % If something is a package, it has only one version and that must be a
 % declared version.
-1 { version(Package, Version) : version_declared(Package, Version) } 1
- :- node(Package), error("Each node must have exactly one version").
-
-% A virtual package may have or not a version, but never has more than one
-:- virtual_node(Package), 2 { version(Package, _) }.
+{ version(Package, Version) : version_declared(Package, Version) }
+ :- node(Package).
```

Comment:
comment above this needs adjusting to say that this tells the concretizer to choose a version, but we infer errors if there's more or less than one chosen version.